lisa/sut_orchestrator/azure/notifiers.py

```diff
@@ -0,0 +1,180 @@
+# Copyright (c) Microsoft Corporation.
+# Licensed under the MIT license.
+
+import json
+from dataclasses import dataclass, field
+from typing import Any, Dict, List, Set, Type
+
+from dataclasses_json import dataclass_json
+
+from lisa import messages, notifier, schema
+from lisa.environment import EnvironmentMessage, EnvironmentStatus
+from lisa.util import LisaException, constants
+from lisa.util.logger import Logger
+
+from .. import AZURE
+
+
+@dataclass
+class VmsizeBasicInfo:
+    name: str = ""
+    vm_family: str = ""
+    core_count: int = 0
+
+
+@dataclass
+class EnvironmentVmsizeInfo:
+    name: str
+    status: EnvironmentStatus = EnvironmentStatus.New
+    vm_sizes: List[VmsizeBasicInfo] = field(default_factory=list)
+
+
+@dataclass_json()
+@dataclass
+class AzureNotifierSchema(schema.Notifier):
+    # fields to query test failure and test cases
+    file_name: str = "azure_usage_counter.json"
+
+
+class AzureNotifier(notifier.Notifier):
+    """
+    This notifier is used to collect the core usage information, which can then guide
+    the quota request in the subscription.
+    """
+
+    def __init__(self, runbook: AzureNotifierSchema) -> None:
+        notifier.Notifier.__init__(self, runbook)
+        self._file_name = runbook.file_name
+
+    @classmethod
+    def type_name(cls) -> str:
+        return "azure_usage_counter"
+
+    @classmethod
+    def type_schema(cls) -> Type[schema.TypedSchema]:
+        return AzureNotifierSchema
+
+    def _received_message(self, message: messages.MessageBase) -> None:
+        if isinstance(message, EnvironmentMessage):
+            self._process_environment_message(message)
+        else:
+            raise LisaException(f"unsupported message received, {type(message)}")
+
+    def _subscribed_message_type(self) -> List[Type[messages.MessageBase]]:
+        return [EnvironmentMessage]
+
+    def _initialize(self, *args: Any, **kwargs: Any) -> None:
+        env_path = constants.RUN_LOCAL_LOG_PATH / "environments"
+        env_path.mkdir(exist_ok=True, parents=True)
+        self._file_path = env_path / f"{self._file_name}.json"
```

Comment:
There is an extra ".json" with default value in the runbook schema.

Reply:
Done